geotrek/common/parsers.py

```diff
@@ -1477,6 +1481,34 @@ def next_row(self):
 
             self.next_url = self.root['next']
 
+    def get_sources_extra_fields(self):
+        response = self.request_or_retry(f"{self.url}/api/v2/source/")
+        create = self.field_options['source'].get("create", False)
+        if create:
+            for result in response.json()['results']:
+                name = result['name']
+                pictogram_url = result['pictogram']
+                website = result['website']
+                source, created = RecordSource.objects.update_or_create(**{'name': name}, defaults={'website': website})
+                if created:
+                    self.add_warning(_(f"Source '{name}' did not exist in Geotrek-Admin and was automatically created"))
+                if not pictogram_url and source.pictogram:
+                    source.pictogram.delete()
+                elif pictogram_url:
+                    pictogram_filename = os.path.basename(pictogram_url)
```

Comment:
I do not understand the URL to filename conversion. When I test locally:

```
>>> os.path.basename("https://foo.bar/hello/world/")
''
```

Reply:
Pictogram URLs include the filename:
```
>>> os.path.basename("https://foo.bar/hello/world/afile.txt")
'afile.txt'
```

Reply:
Alright :+1: